src/main/java/com/puppycrawl/tools/checkstyle/xpath/XpathQueryGenerator.java

```diff
@@ -332,4 +332,75 @@ private boolean isMatchingByLineAndColumnAndTokenType(DetailAST ast) {
                 && expandedTabColumn(ast) == columnNumber
                 && (tokenType == 0 || tokenType == ast.getType());
     }
+
+    /**
+     * Escape &lt;, &gt;, &amp;, &#39; and &quot; as their entities.
+     * Custom method for Xpath generation to maintain compatibility
+     * with Saxon and encoding outside Ascii range characters.
+     *
+     * <p>According to Saxon documentation in http://saxon.sourceforge.net/saxon7.1/expressions.html
+     * From Saxon 7.1, string delimiters can be doubled within the string to represent
+     * the delimiter itself: for example select='"He said, ""Go!"""'.
+     *
+     * <p>Guava cannot be used as guava is not able to encode outside ASCII Character them till now
+     * as there documentation states "Currently the escapes provided by this class do not escape
+     * any characters outside the ASCII character range". Also, Guava encoding does not meet our
+     * requirements like double encoding for apos, removed slashes which are basic requirements
+     * for Saxon to decode.
+     *
+     * @param value the value to escape.
+     * @return the escaped value if necessary.
+     */
+    public static String encode(String value) {
```

Comment:
We have raletively easy way to test this method by input files.
Please make it private.

Reply:
Sir, I will not be able to do changes till the 8th of march due to my examinations. So, please wait till then. Also, please answer this [#9247 (comment)](https://github.com/checkstyle/checkstyle/pull/9247#discussion_r582811838). It would be helpful.

Reply:
Done